src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -326,6 +326,34 @@ public void AssemblyLoading_MultipleVersions_MultipleEqualMatches()
             }
         }
 
+
+        [Fact]
+        public void AssemblyLoading_MultipleVersions_MultipleVersionsOfSameAnalyzerItself()
+        {
+            StringBuilder sb = new StringBuilder();
+
+            var loader = new DefaultAnalyzerAssemblyLoader();
+            loader.AddDependencyLocation(_testFixture.Delta2.Path);
+            loader.AddDependencyLocation(_testFixture.Delta2B.Path);
+
+            Assembly delta2 = loader.LoadFromPath(_testFixture.Delta2.Path);
+            Assembly delta2B = loader.LoadFromPath(_testFixture.Delta2B.Path);
+
+            // 2B or not 2B? That is the question...that depends on whether we're on .NET Core or not.
+
+#if NETCOREAPP
+
+            // On Core, we're able to load both of these into separate AssemblyLoadContexts.
+            Assert.NotEqual(delta2B.Location, delta2.Location);
```

Comment:
Should we additionally test that these locations are equal to `_testFixture.Delta2B.Path` and `_testFixture.Delta2.Path` respectively?

Reply:
Sure, that'd be good to add.